BRANCHES.md

```diff
@@ -28,6 +28,15 @@ undocumented features, then you should be able to re-compile it without
 modification with any later release x.y'.z' with the same major version
 number, and your code will still build, be secure, and work.
 
+Note that new releases of Mbed TLS may extend the API. Here are some example of changes that are common in minor releases of Mbed TLS, and are not considered compatibility breaks:
+
+* Adding or reordering fields in a structure or union.
+* Removing a field from a structure, unless the field is documented as public.
+* Adding items to an enum.
```

Comment:
What about changing items' values?

Reply:
Do we have any enums that are functionally dependant on the actual values of the enum? Presumably such items would have an explanatory comment next to them otherwise it becomes a maintenance nightmare. Perhaps we should caveat that adding items is only ok if changing the values of an enum has no effect.

Reply:
Inside the library, I don't think there's any behavior that depends on the numerical values. But it could matter for people who write FFI and replicate enums as constant in the high-level language.

I think the rules should be the same for enum values and for numerical `#define`s. Many enum-like values are defines because we want to control the size of the storage or because the type is open-ended in the module that defines it.

Reply:
Definitely agree on the enums where they are used as #defines, I was pondering the case where adding an enum item to the middle of an enum causes renumbering of items after it, and some code is dependant on items having given values. That would not, admittedly be great code though.

Reply:
I think the case Paul mentions is an ABI break, obviously, but not an API break, on the basis that the API is defined using the enum names, not their values. So I don't think it would be completely wrong to add a bullet for "Changing existing enum values" - but that's not a common thing for us to do (as per the note preceding the list) so I would suggest we don't mention it.